tests/runner.py

```diff
@@ -521,7 +521,10 @@ def assertTextDataContained(self, text1, text2):
   def assertContained(self, values, string, additional_info=''):
     if type(values) not in [list, tuple]: values = [values]
     for value in values:
-      if not isinstance(value, bytes): string = string.decode('UTF-8') # If we have any non-ASCII chars in the expected string, treat the test string from ASCII as UTF8 as well.
+      if str is bytes and type(value) is unicode: 
```

Comment:
is it worth making a helper function for this case too?

Reply:
Slightly wired logic here, we may do values = map(asstr, values) instead.